### PR TITLE
Standardise report structure using a Report class

### PR DESCRIPTION
**Pull request by digitalronin**
Create a Report class, rather than having scripts build their own,
ad hoc reports

### CODE REVIEW

File: spec/report_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  it 'accepts a repo' do
+    expect { described_class.new(repo: repo) }.to_not raise_error
+  end
+
+  it 'accepts a user & repo name' do
+    expect { described_class.new(user: 'some-user', repo: 'some-repo') }.to_not raise_error
+  end
+
+  it 'raises an exception when required parameters are not supplied' do
+    expect { described_class.new(foo: 'bar') }.to raise_error(KeyError)
```
Comment (SteveMarshall) [inline spec/report_spec.rb]: 👌

**Comment by digitalronin** [comment]
I've rebased from master, but also mocked the requirement of a GH_TOKEN env. var. in the specs, so this need re-reviewing.

File: spec/report_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+require_relative 'spec_helper'
+
+describe RepoAudit::Report do
+  let(:repo) { double(Github::Mash, { full_name: 'some-user/not-a-real-repo' }) }
+  let(:report) { described_class.new(repo: repo) }
+  let(:checker) { double(run: nil) }
+
+  before do
+    allow(RepoAudit::LicenseChecker).to receive(:new).with(repo).and_return(checker)
+    allow(ENV).to receive(:fetch).with('GH_TOKEN').and_return('dummy-token')
```
Comment (SteveMarshall) [inline spec/report_spec.rb]: Hmmm… I wonder if the token should be passed into `initialize()`. Accessing `ENV` directly feels a little spooky-action-y to me? I'm fine with it for this PR, but might be worth changing later maybe…

Comment (digitalronin) [inline spec/report_spec.rb]: Possibly. It always feels a bit weird, to me, to supply tokens as command-line arguments, though. Also, using ENV.fetch makes it very simple to blow things up in an easily understandable way. If you want to have the same effect with a token passed to an initialiser, IMO it makes it a (very) little bit harder to explain to the user what they're supposed to do about it.

Comment (SteveMarshall) [inline spec/report_spec.rb]: Oh, I'm not saying it shouldn't use an env var, just that the `ENV.fetch` maybe shouldn't be in `RepoAudit::Report.fetch_repo()`. (Rather, that we should pass the GitHub token to `RepoAudit::Report.initializer()` having `ENV.fetch`'d in `bin/whatever.rb`.)

Comment (digitalronin) [inline spec/report_spec.rb]: Ah, got it. I misunderstood. Yes, good idea. I'll add that as an issue.

**Pull request merged**